public/app/plugins/panel/barchart/utils.ts

```diff
@@ -294,97 +303,129 @@ export function preparePlotFrame(data: DataFrame[]) {
 }
 
 /** @internal */
-export function prepareGraphableFrames(
+export function prepareBarChartDisplayValues(
   series: DataFrame[],
   theme: GrafanaTheme2,
   options: BarChartOptions
-): { frames?: DataFrame[]; warn?: string } {
+): BarChartDisplayValues {
   if (!series?.length) {
-    return { warn: 'No data in response' };
+    return { warn: 'No data in response' } as BarChartDisplayValues;
   }
 
-  const frames: DataFrame[] = [];
-  const firstFrame = series[0];
-
-  if (!firstFrame.fields.some((f) => f.type === FieldType.string)) {
-    return {
-      warn: 'Bar charts requires a string field',
-    };
+  // Bar chart requires a single frame
+  const frame = series.length === 1 ? series[0] : outerJoinDataFrames({ frames: series, enforceSort: false });
```

Comment:
is the current behavior joining deep down?  This will just do it at the top instead